redfish-core/lib/pcie.hpp

```diff
@@ -423,188 +423,193 @@ inline void requestRoutesSystemPCIeDeviceCollection(App& app)
-                            asyncResp->res.jsonValue["Manufacturer"] =
-                                *property;
-                        }
-
-                        if (std::string* property = std::get_if<std::string>(
-                                &pcieDevProperties["DeviceType"]);
-                            property)
-                        {
-                            if (!property->empty())
-                            {
-                                asyncResp->res.jsonValue["DeviceType"] =
-                                    *property;
-                            }
-                        }
-
-                        if (std::string* property = std::get_if<std::string>(
-                                &pcieDevProperties["GenerationInUse"]);
-                            property)
-                        {
-                            std::string generation =
-                                analysisGeneration(*property);
-                            if (!generation.empty())
-                            {
-                                asyncResp->res
-                                    .jsonValue["PCIeInterface"]["PCIeType"] =
-                                    generation;
-                            }
-                        }
+                BMCWEB_LOG_DEBUG
+                    << "failed to get PCIe Device properties ec: " << ec.value()
+                    << ": " << ec.message();
+                if (ec.value() ==
+                    boost::system::linux_error::bad_request_descriptor)
+                {
+                    messages::resourceNotFound(asyncResp->res, "PCIeDevice",
+                                               device);
+                }
+                else
+                {
+                    messages::internalError(asyncResp->res);
+                }
+                return;
+            }
 
-                        if (std::string* property = std::get_if<std::string>(
-                                &pcieDevProperties["PartNumber"]);
-                            property)
-                        {
-                            if (!property->empty())
-                            {
-                                asyncResp->res.jsonValue["PartNumber"] =
-                                    *property;
-                            }
-                        }
+            asyncResp->res.jsonValue = {
+                {"@odata.type", "#PCIeDevice.v1_9_0.PCIeDevice"},
+                {"@odata.id",
+                 "/redfish/v1/Systems/system/PCIeDevices/" + device},
+                {"Name", "PCIe Device"},
+                {"Id", device}};
+            std::vector<std::string> splitText;
+            boost::split(splitText, path, [](char c) { return c == '/'; });
+            auto it = std::find_if(splitText.begin(), splitText.end(),
+                                   [](const std::string& data) {
+                                       return data.find("chassis") !=
```

Comment:
Would it be better to match on "Chassis" and then take the child? 
I know the mex drawer can have PCIeSlots as well and what happens if the name for the mex drawer doesn't have "chassis" substring in it?